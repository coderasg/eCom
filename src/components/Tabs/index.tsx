import React, { ReactNode } from 'react';

import styles from './tabs.module.scss';

type Props = {
  children?: ReactNode;
  direction?: 'VERTICAL' | 'HORIZONTAL'; //TODO use typescript enum but showing eslint error of unused var
};

function Tabs(props: Props) {
  const { children, direction = 'HORIZONTAL' } = props;

  return (
    <div
      data-testid="Tabs-wrapper"
      className={
        direction === 'VERTICAL' ? styles.verticalTabs : styles.horizontalTabs
      }
    >
      {children}
    </div>
  );
}

export default Tabs;
