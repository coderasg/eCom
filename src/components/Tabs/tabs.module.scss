@use "./../../variables";


.horizontalTabs {
  display: flex;
  flex-grow: 1;
  justify-content: flex-start;
  background-color: white;
  padding: variables.$paddingMid;;
  overflow-x: auto;
  overflow-y: hidden;
  position: sticky;
  top: 51px;
}
