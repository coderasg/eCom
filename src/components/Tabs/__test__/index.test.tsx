import React from 'react';
import renderer from 'react-test-renderer';

import Tabs from '../index';

describe('(Component) Tabs', () => {
  test('should pass snapshot test', () => {
    const wrapper = renderer.create(<Tabs>{'aesr'}</Tabs>);
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
