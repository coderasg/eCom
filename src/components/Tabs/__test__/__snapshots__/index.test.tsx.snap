// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) Tabs should pass snapshot test 1`] = `
<div
  data-testid="Tabs-wrapper"
>
  aesr
</div>
`;
